core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -27,15 +30,19 @@ sealed abstract class Expr[-I, +O : OP, OP[_]](val name: String) {
   /**
     * Holds any [[Debugging]] hook to be run while running this expression.
     *
-    * You should favor using the `.debug` extension method provided by the [[Expr.DebugSyntax]]
+    * You should favor using the `.debug` extension method provided by the [[DebugArgs.Attacher]] using an
+    * implicit conversion of this [[Expr]].
     *
     * @note Only one hook can be attached to an expression. The last one attached wins.
     *       TODO: Remove this limitation by chaining debug functions together.
     *
     * @note As a top-level [[Expr]] method, this is not very useful, as you cannot invoke it, however, you can
     *       print or inspect it for information at runtime; and placing it here prevents me from forgetting to
-    *       add this to subclasses. The types are refined when attaching the debug hooks using the [[DebugArgs]]
-    *       compiler trick.
+    *       add this to subclasses.
+    *
+    * @see [[Debugging]] for more details on how type checking is handled.
+    *
+    * TODO: Should this be made package private?
```

Comment:
I think I probably should make the debugging field more private. It would be a confusing interface if you aren't working from within the library.